lib/java/persistance-dsl/src/main/java/org/enso/persist/impl/PersistableProcessor.java

```diff
@@ -274,12 +274,12 @@ public int compare(Object a, Object b) {
     return true;
   }
 
-  private int countSeq(List<? extends VariableElement> parameters) {
+  private int countInlineRef(List<? extends VariableElement> parameters) {
     var tu = processingEnv.getTypeUtils();
     var cnt = 0;
     for (var p : parameters) {
       var type = tu.asElement(tu.erasure(p.asType()));
-      if (type != null && type.getSimpleName().toString().equals("Seq")) {
+      if (type != null && type.getSimpleName().toString().equals("InlineReference")) {
         cnt++;
       }
```

Comment:
IMHO this heuristic is much safer once it relies on a special type instead of a general-purpose one.